Core/include/Acts/Detector/DetectorVolumeBuilder.hpp

```diff
@@ -0,0 +1,78 @@
+// This file is part of the Acts project.
+//
+// Copyright (C) 2022-2023 CERN for the benefit of the Acts project
+//
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+#pragma once
+
+#include "Acts/Detector/interface/IDetectorComponentBuilder.hpp"
+#include "Acts/Detector/interface/IExternalStructureBuilder.hpp"
+#include "Acts/Detector/interface/IInternalStructureBuilder.hpp"
+#include "Acts/Utilities/Logger.hpp"
+
+#include <memory>
+#include <string>
+
+namespace Acts {
+namespace Experimental {
+
+/// @brief A generic detector volume builder that uses
+/// an external builder for shape and portals and an internal
+/// structure builder for volume internals.
```

Comment:
```suggestion
/// structure builder for internal volumes.
```

Reply:
Thi is actually for internals of the volume.